### DISCUSSION TITLE
How to Use Systembackdrop with OnTargetConnected ?

### DISCUSSION THREAD

**Discussion by ghost1372** [microsoft/WindowsAppSDK#4103] [Q&A]
i am trying to use Mica System backdrop with OnTargetConnected  method, but there is some issues, for example when i change my application theme, application crash! how i can implement this in correct way?
i used codes in docs:

```
public class MicaLightSystemBackdrop : SystemBackdrop
{
    MicaController micaController;

    protected override void OnTargetConnected(ICompositionSupportsSystemBackdrop connectedTarget, XamlRoot xamlRoot)
    {
        base.OnTargetConnected(connectedTarget, xamlRoot);

        if (micaController is not null)
        {
            throw new Exception("This controller cannot be shared");
        }

        micaController = new MicaController();
        //_ = GetDefaultSystemBackdropConfiguration(connectedTarget, xamlRoot);

        micaController.AddSystemBackdropTarget(connectedTarget);
    }

    protected override void OnTargetDisconnected(ICompositionSupportsSystemBackdrop disconnectedTarget)
    {
        base.OnTargetDisconnected(disconnectedTarget);

        micaController.RemoveSystemBackdropTarget(disconnectedTarget);
        micaController = null;
    }
}
``` 
after some test i used OnDefaultSystemBackdropConfigurationChanged and i can fix crash by checking null value:

```
protected override void OnDefaultSystemBackdropConfigurationChanged(ICompositionSupportsSystemBackdrop target, XamlRoot xamlRoot)
    {
        if(target != null){
base.Default....
}
    }
```

however this only fix crash but does not fix application theme and backdrop

**Comment by Gaoyifei1011** [comment]
Maybe you could see reference https://github.com/Gaoyifei1011/GetStoreApp/blob/main/GetStoreApp/Extensions/Backdrop/MaterialBackdrop.cs

**Reply by ghost1372** [reply]
tnx, i used your codes and again i get exception. (only if i change Application theme), 
if i add following code, i can bypass app crash:
    
```
protected override void OnDefaultSystemBackdropConfigurationChanged(ICompositionSupportsSystemBackdrop target, XamlRoot xamlRoot)
    {
        if (target != null)
        {
            
        }
    }

```

however there is another issue, if i change TintColor, backdrop theme does not change if i change application theme

**Comment by Gaoyifei1011** [comment]
> tnx, i used your codes and again i get exception. (only if i change Application theme), if i add following code, i can bypass app crash:
> 
> ```
> protected override void OnDefaultSystemBackdropConfigurationChanged(ICompositionSupportsSystemBackdrop target, XamlRoot xamlRoot)
>     {
>         if (target != null)
>         {
>             
>         }
>     }
> ```
> 
> however there is another issue, if i change TintColor, backdrop theme does not change if i change application theme

I'm sorry, I may not have understood you earlier, but now I understand. I have encountered this problem before, and this is my temporary solution
![image](https://github.com/microsoft/WindowsAppSDK/assets/49179966/a643d836-d3c4-4650-be94-4ebff5cffc32)
https://github.com/Gaoyifei1011/GetStoreApp/blob/main/GetStoreApp/WinUIApp.xaml.cs

**Reply by ghost1372** [reply]
your solution does not work for me, i fixed app crash by overriding OnDefaultSystemBackdropConfigurationChanged, but i have still issue with changing TintColor and Changing App Theme

**Reply by Gaoyifei1011** [reply]
![image](https://github.com/microsoft/WindowsAppSDK/assets/49179966/352ceedc-3809-40ee-8644-cd06d56a3dc0)
Thank you very much for providing the method, when I added such a judgment code against my custom SystemBackdrop, I did not actively raise the UnhandledException when I modified the theme.

I checked the your code, I found you is according to the advice of the official document for Microsoft operating, you didn't pass the GetDefaultSystemBackdropConfiguration method to get SystemBackdropConfiguration object, But through initializes a new SystemBackdropConfiguration object. This new SystemBackdropConfiguration object when changing themes need to manually monitor application theme of change, the default that don't need to
